core/iwasm/libraries/libc-uvwasi/libc_uvwasi.cmake

```diff
@@ -9,36 +9,52 @@ add_definitions (-DWASM_ENABLE_LIBC_WASI=1 -DWASM_ENABLE_UVWASI=1)
 
 include(FetchContent)
 
+# Point CMake at the custom modules to find libuv and uvwasi
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")
+
 ## libuv
-FetchContent_Declare(
-    libuv
-    GIT_REPOSITORY https://github.com/libuv/libuv.git
-    GIT_TAG ${LIBUV_VERSION}
-)
-FetchContent_GetProperties(libuv)
-if (NOT libuv_POPULATED)
-    message("-- Fetching libuv ..")
-    FetchContent_Populate(libuv)
-    include_directories("${libuv_SOURCE_DIR}/include")
-    add_subdirectory(${libuv_SOURCE_DIR} ${libuv_BINARY_DIR} EXCLUDE_FROM_ALL)
-    set (UV_A_LIBS uv_a)
-    set_target_properties(uv_a PROPERTIES POSITION_INDEPENDENT_CODE 1)
+find_package(LIBUV QUIET)
+if (LIBUV_FOUND)
+    include_directories(${LIBUV_INCLUDE_DIR})
+else()
+    FetchContent_Declare(
+        libuv
+        GIT_REPOSITORY https://github.com/libuv/libuv.git
+        GIT_TAG ${LIBUV_VERSION}
+    )
+    FetchContent_GetProperties(libuv)
+    if (NOT libuv_POPULATED)
+        message("-- Fetching libuv ..")
+        FetchContent_Populate(libuv)
+        include_directories("${libuv_SOURCE_DIR}/include")
+        add_subdirectory(${libuv_SOURCE_DIR} ${libuv_BINARY_DIR} EXCLUDE_FROM_ALL)
+        set (LIBUV_LIBRARIES uv_a)
+        set_target_properties(uv_a PROPERTIES POSITION_INDEPENDENT_CODE 1)
+    endif()
 endif()
 
 ## uvwasi
-FetchContent_Declare(
-    uvwasi
-    GIT_REPOSITORY https://github.com/nodejs/uvwasi.git
-    GIT_TAG main
-)
-FetchContent_GetProperties(uvwasi)
-if (NOT uvwasi_POPULATED)
-    message("-- Fetching uvwasi ..")
-    FetchContent_Populate(uvwasi)
-    include_directories("${uvwasi_SOURCE_DIR}/include")
-    add_subdirectory(${uvwasi_SOURCE_DIR} ${uvwasi_BINARY_DIR} EXCLUDE_FROM_ALL)
+find_package(UVWASI QUIET)
+if (UVWASI_FOUND)
+    include_directories(${UVWASI_INCLUDE_DIR})
+else()
+    FetchContent_Declare(
+        uvwasi
+        GIT_REPOSITORY https://github.com/nodejs/uvwasi.git
+        GIT_TAG main
+    )
+    FetchContent_GetProperties(uvwasi)
+    if (NOT uvwasi_POPULATED)
+        message("-- Fetching uvwasi ..")
+        FetchContent_Populate(uvwasi)
+        include_directories("${uvwasi_SOURCE_DIR}/include")
+        add_subdirectory(${uvwasi_SOURCE_DIR} ${uvwasi_BINARY_DIR} EXCLUDE_FROM_ALL)
+        set (UVWASI_LIBRARIES uvwasi_a)
```

Comment:
typo: add them

Reply:
It looks like this was already added for uv_a so I just added it to `uvwasi_a`

Reply:
The issue in my machine is that package libuv1-dev is installed while no uvwasi related package is installed, and `find_package(LIBUV QUIET)` returns true but uvwasi must be compiled from source code. And when downloading and compiling uvwasi, `it also downloads, compiles and links libuv (static lib)` instead of linking the libuv.so from the installed package. And here the compiler reports error when linking libuv.a. I don't know how to let uvwasi use the installed libuv package directly, if we can, it is unnecessary to add the line. Otherwise, had better add the line. Any idea for it? Thanks.

Reply:
>  I don't know how to let uvwasi use the installed libuv package directly, if we can, it is unnecessary to add the line.

Sorry for the confusion. My PR at https://github.com/nodejs/uvwasi/pull/210 was required to have uvwasi use the same global version. I've now merged the PR into the main branch so you should be able to build with the new source code and it will link the global libuv.

Reply:
OK, thanks a lot. I tested again, it links the correct libuv now. An issue is in Ubuntu-20.04, when package libuv1-dev is installed, compile warning and link error occur:
```bash
product-mini/platforms/linux/build/_deps/uvwasi-src/src/uvwasi.c:1743:7: warning: implicit declaration of function ‘uv_fs_lutime’; did you mean ‘uv_fs_futime’? [-Wimplicit-function-declaration]
 1743 |   r = uv_fs_lutime(NULL, &req, resolved_path, atim, mtim, NULL);
```
```bash
[ 98%] Linking C executable iwasm
/usr/local/bin/ld: _deps/uvwasi-build/libuvwasi_a.a(uvwasi.c.o): in function `uvwasi_path_filestat_set_times':
uvwasi.c:(.text+0x293f): undefined reference to `uv_fs_lutime'
```
Note that the warning/error isn't reported in Ubuntu-22.04, it works fine in Ubuntu-22.04.
I think it is due to the installed libuv.so doesn't have the `uv_fs_lutime` function. It is not the issue of this PR, let's merge it first. And it would be good if we can resolve the issue, e.g. modifying uvwasi by using `uv_fs_futime`?

Reply:
That sounds good. I don't have experience with the libuv function mentioned, but I can raise the issue for the other members of the uvwasi team with more libuv experience.

Reply:
That issue has been raised at https://github.com/nodejs/uvwasi/issues/213